x/crosschain/keeper/cctx.go

```diff
@@ -47,6 +30,23 @@ func (k Keeper) SetCctxAndNonceToCctxAndInTxHashToCctx(ctx sdk.Context, cctx typ
 			Tss:       tss.TssPubkey,
 		})
 	}
+
+	k.SetCrossChainTx(ctx, cctx)
+	//set mapping inTxHash -> cctxIndex
+	in, _ := k.GetInTxHashToCctx(ctx, cctx.InboundTxParams.InboundTxObservedHash)
+	in.InTxHash = cctx.InboundTxParams.InboundTxObservedHash
+	found = false
+	for _, cctxIndex := range in.CctxIndex {
+		if cctxIndex == cctx.Index {
+			found = true
+			break
+		}
+	}
+	if !found {
+		in.CctxIndex = append(in.CctxIndex, cctx.Index)
+	}
+	k.SetInTxHashToCctx(ctx, in)
+
```

Comment:
Just wondering, any particular reason this logic is moved?

Reply:
Get tss is the only function that returns , so I moved it up .
We should probably return and error here and handle it from the calling function , Will add an issue to track it